controllers/helper.go

```diff
@@ -302,17 +365,7 @@ func syncDaemonSet(ctx context.Context, client k8sclient.Client, scheme *runtime
 
 		if equality.Semantic.DeepEqual(in.OwnerReferences, ds.OwnerReferences) &&
 			equality.Semantic.DeepDerivative(in.Spec, ds.Spec) {
-			// DeepDerivative has issue detecting nodeAffinity change
-			// https://bugzilla.redhat.com/show_bug.cgi?id=1914066
-			// DeepDerivative doesn't detect changes in containers args section
-			// This code should be fixed both with NodeAffinity comparation
-			if equality.Semantic.DeepEqual(in.Spec.Template.Spec.Affinity.NodeAffinity,
-				ds.Spec.Template.Spec.Affinity.NodeAffinity) &&
-				equality.Semantic.DeepEqual(in.Spec.Template.Spec.Containers[0].Args,
-					ds.Spec.Template.Spec.Containers[0].Args) {
-				logger.V(1).Info("Daemonset spec did not change, not updating")
```

Comment:
I think the log is still valid before `return nil`

Reply:
I didn't remove I just move the functions to the right file.

before this change functions in helpers called functions in the controller file that breaks the point of helpers

Reply:
I meant this line

`logger.V(1).Info("Daemonset spec did not change, not updating")`

Reply:
ohhhh you are right! sorry stupid me :(